### PR TITLE
Video Optimization: Telemetry

### LINKED ISSUE #5990: Video Optimization: Telemetry
## Feature Description

We'd want to know how often videos get transcoded, and how long it takes.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* [ ] Track media upload events themselves
* [ ] Track transcoding events themselves
* [x] Track errors
* [ ] Track timing

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add some basic tracking to media upload and transcoding. 

## Relevant Technical Choices

I made sure that timers finished, by adding to them in places in code that are wrapped in try/catches and finishing on finally. 

The placement of the file upload, where it is because multiple files can be uploaded at once.  

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #5990

### CODE REVIEW

File: assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js [hunk 1]
```diff
@@ -89,12 +89,21 @@ function useUploadMedia({ media, setMedia }) {
 
       try {
         const uploadedFiles = await Promise.all(
-          localFiles.map(async (localFile) => ({
-            ...localFile,
-            fileUploaded: getResourceFromAttachment(
+          localFiles.map(async (localFile) => {
```
Comment (spacedmonkey) [inline assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js:96]: I didn't add `trackEvent`, as I didn't see the point. Felt like I would be repeating events. But I can add this if required.

File: assets/src/edit-story/app/uploader/useUploader.js [hunk 2]
```diff
@@ -118,6 +118,11 @@ function useUploader() {
       };
 
       if (canTranscodeFile(file)) {
+        const trackTiming = getTimeTracker(
```
Comment (spacedmonkey) [inline assets/src/edit-story/app/uploader/useUploader.js]: See https://github.com/google/web-stories-wp/pull/6188/files#r570318142

File: assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js:96]: They're slightly different in analytics, so please also add `trackEvent` and include the mime type as I described in the ticket.

I want to have both first, then I can see in analytics how well it works and whether it's indeed redundant or not.

Comment (spacedmonkey) [reply assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js:96]: Fair enough. Added a fix.

File: assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js [hunk 3]
```diff
@@ -89,12 +89,24 @@ function useUploadMedia({ media, setMedia }) {
...
+            trackEvent('upload_media', 'editor', '', '', {
+              type: localFile.file.type,
```
Comment (spacedmonkey) [inline assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js]: I think we should add file size as well here.

Comment (swissspidy) [reply assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js]: SGTM

Let's use `file_type` and `file_size` as keys then.

Comment (spacedmonkey) [reply assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js]: Fixed https://github.com/google/web-stories-wp/pull/6188/commits/f7d8a62cbb850d53ff2738c5aad84652eb34f302

File: assets/src/edit-story/app/uploader/useUploader.js [hunk 4]
```diff
@@ -149,13 +153,25 @@ function useUploader() {
...
+
       // Transcoding is enabled, let's give it a try!
       try {
         // TODO: Only transcode & optimize video if needed (criteria TBD).
         const newFile = await transcodeVideo(file);
+        trackTiming();
         additionalData.media_source = 'video-optimization';
         return uploadMedia(newFile, additionalData);
       } catch (err) {
+        trackTiming();
```
Comment (swissspidy) [inline assets/src/edit-story/app/uploader/useUploader.js:174]: Let's just write this once in a `finally {}` block at the end.

**Review by swissspidy — CHANGES_REQUESTED** [review]
just one nit

File: assets/src/edit-story/app/uploader/useUploader.js (hunk 4 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/uploader/useUploader.js:174]: I did originally use a finally [here](https://github.com/google/web-stories-wp/pull/6188/commits/f13f6b91dc954724c9d224d4e7cb2a00f44cbdef). But changed as the linter didn't like it. The issue is there is a return on line 169. I can revert the change and ignore the lint, but I think it is correct.

Comment (swissspidy) [reply assets/src/edit-story/app/uploader/useUploader.js:174]: Ah, didn't see the return there. Yes you're right, let's leave it then.

**Review by swissspidy — APPROVED** [review]

**Comment by spacedmonkey** [comment]
Does this need QA instructions?

**Comment by swissspidy** [comment]
> Does this need QA instructions?

Not needed as it's not a user-facing change that requires QA.

**Pull request merged**